src/views/student/CourseInfoPage.vue

```diff
@@ -1,98 +1,70 @@
 <template>
-    <div v-if="busy > 0" class="flex h-screen">
-        <div class="m-auto">
-            <loading-component />
-        </div>
-    </div>
-    <div v-else class="w-full lg:mt-20 mt-8 bg-gray-300 mx-auto h-screen">
-        <button id="back" class="flex items-center mb-4 navigation-link" @click="back">
-            <i class="fas text-xl fa-chevron-left"></i>
-            <span class="font-bold text-sm ml-1">Back</span>
-        </button>
-
-        <h1 class="text-2xl font-medium text-gray-700 mb-8">{{ course.courseName }}</h1>
-        <div>
-            <lecturer-section :lecturer="lecturerName" :lecturer-id="course.lecturerId" />
-            <basics-section
+    <base-view>
+        <loading-spinner v-if="isLoading" title="Loading Course" />
+        <div v-else>
+            <section-header :title="course.courseName" />
+            <presenter-section :lecturer="lecturer" />
+            <info-section
                 :name="course.courseName"
                 :type="course.courseType"
                 :language="course.courseLanguage"
                 :ects="course.ects"
                 :description="course.courseDescription"
             />
-            <module-section v-model:selected="selectedModule" :module-ids="course.moduleIds" :registered="isRegistered" />
+            <module-section :selected="selectedModule" :module-ids="course.moduleIds" :registered="isRegistered" />
```

Comment:
This does not work properly. One has to select the module twice before it is selected: It has to be v-modeled here, such that the student is able to join the course (the admission needs the module). See disabling the join button with `!valid`

```suggestion
            <module-section v-model:selected="selectedModule" :module-ids="course.moduleIds" :registered="isRegistered" />
```